File: app/src/main/java/com/karthik/splash/Modules/SplashAppModule.java
package com.karthik.splash.Modules;

import android.content.Context;

import com.karthik.splash.Storage.Cache;

import dagger.Module;
import dagger.Provides;

/**
 * Created by karthikrk on 12/11/17.
 */
@Module
public class SplashAppModule {
    private Context context;

    public SplashAppModule(Context context){
        this.context = context;
    }

    @Provides
    public Context providesContext(){
        return context;
    }

    @Provides
    public Cache providesCache(Context context){
        return new Cache(context);
    }
}

File: app/src/main/java/com/karthik/splash/root/SplashAppModule.kt
package com.karthik.splash.root

import android.content.Context
import com.karthik.splash.misc.InternetHandler
import com.karthik.splash.storage.MemoryCache
import com.karthik.splash.storage.db.SplashDatabase
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SplashAppModule(private val context: Context, private val splashDatabase: SplashDatabase) {

    @Provides
    fun providesContext() =
            context

    @Provides
    fun providesCache() =
            MemoryCache(context)

    @Singleton
    @Provides
    fun providesDatabaseDao() =
            splashDatabase.getDatabaseDao()

    @Singleton
    @Provides
    fun providesInternetHandler() =
            InternetHandler(context)
}